R/clean_garbage.R

```diff
@@ -0,0 +1,153 @@
+# Declarations used in package check
+globalVariables(
+  names = c('edgelist_dir', 'variable', 'pattern_match', 'type', 'size'),
+  package = 'pixelatorR',
+  add = TRUE
+)
+
+#' Clean up Unlinked Edgelist Directories
+#'
+#' Each time a \code{Seurat} with a \code{CellGraphAssay} or a \code{CellGraphAssay} is created,
+#' renamed, filtered (by cells) or merged, a copy of the associated edgelist is made
+#' in the \code{getOption("pixelatorR.arrow_outdir")} directory. For large data sets,
+#' these edgelist files can use up several gigabytes of disk space. This function
+#' scans through the \code{getOption("pixelatorR.arrow_outdir")} directory and removes
+#' any edgelist files that are not linked to \code{Seurat} objects or \code{CellGraphAssay}s
+#' in the global environment. Note that only global variables which inherits the \code{Seurat}
+#' or \code{CellGraphAssay} class are considered, as well as lists containing these object types.
+#'
+#' @export
+#'
+clean_edgelists_directories <- function () {
+
+  # Get variables in global environment
+  global_variables <- global_env() %>% as.list()
+
+  # Check if there are any Seurat objects or CellGraphAssays
+  global_variable_types <- sapply(names(global_variables), function(nm) {
+
+    x <- global_variables[[nm]]
+
+    if (inherits(x, what = c("Seurat", "CellGraphAssay"))) {
+      # Ignore Seurat objects without CellGraphAssays
+      arrow_dir <- try({
+        ArrowDir(x)
+      }, silent = TRUE)
+      if (inherits(arrow_dir, "try-error")) {
+        return(NULL)
+      } else {
+        return(arrow_dir)
+      }
+    } else if (inherits(x, what = "list")) {
+
+      # Check if any list contains Seurat objects or CellGraphAssays
+      list_arrow_dirs <- lapply(x, function(y) {
+        if (inherits(y, what = c("Seurat", "CellGraphAssay"))) {
+          # Ignore Seurat objects without CellGraphAssays
+          arrow_dir <- try({
+            ArrowDir(y)
+          }, silent = TRUE)
+          if (inherits(arrow_dir, "try-error")) {
+            return(NULL)
+          } else {
+            return(arrow_dir)
+          }
+        } else {
+          return(NULL)
+        }
+      })
+      return(list_arrow_dirs %>% set_names(seq_along(list_arrow_dirs)))
+    } else {
+      return(NULL)
+    }
+  })
+
+  # Unlist and remove NULLs
+  global_variable_types <- unlist(global_variable_types)
+  global_variable_types <- Filter(Negate(is.null), global_variable_types)
+
+  # Convert to tibble
+  if (length(global_variable_types) > 0) {
+    global_variable_types <- tibble(variable = names(global_variable_types), edgelist_dir = global_variable_types)
+  }
+
+  # Get command log
+  command_log <- getOption("pixelatorR.edgelist_copies")
+  if (length(command_log) > 0) {
```

Comment:
Do you want to check the number of rows in the log and the `global_variable_types`? In that case `nrow` should be used since `length` returns the number of columns for data frames.